src/Matrix.hs

```diff
@@ -30,8 +32,8 @@ module Matrix
 
 import Data.Maybe (fromMaybe)
 import qualified Data.List as DL (transpose)
-import qualified Data.Vector as DVB (Vector(..))
-import qualified Data.Vector.Unboxed as DV (Vector(..), (!), and, backpermute, fromList, generate, length, map, maximum, minimum, toList, zipWith, empty)
+import qualified Data.Vector as DVB (Vector(..), toList)
+import qualified Data.Vector.Unboxed as DV (Vector(..), (++), (!), and, backpermute, fromList, generate, length, map, maximum, minimum, toList, zipWith, empty, take, concat)
```

Comment:
Hmmm... `Data.Vector as DVB` and `Data.Vector.Unboxed as DV` seems a lazzy and utterly ugly hack to make things work without proper naming...

Reply:
A bit, but it also avoid having a huge diff with `DV` -> `DVU`, also the Unboxed is now our main/only type of vector.